cylc/flow/graph_parser.py

```diff
@@ -736,6 +743,11 @@ def _set_output_opt(
         if suicide:
             return
 
+        # TODO: OS: "This is now covered by the optional outputs stuff"
+        if output == TASK_OUTPUT_EXPIRED and not optional:
+            raise GraphParseError(
+                f"Expired-output {name}:{output} must be optional")
+
```

Comment:
I added this in with the logic that checks that failed is optional if succeeded is.

Reply:
OK. (That was a reminder to me to check if it was in already or in your extension PR).

Reply:
Do you want me to leave this here for now?

Reply:
Yeah, leave this be, can remove the TODO.